app/src/main/java/com/huji/foodtricks/buddies/EventAttendanceProvider.java

```diff
@@ -22,6 +20,19 @@
     private HashMap<String, String> tentatives;
     private HashMap<String, String> nonResponsive;
 
+    public enum RSVP {ATTENDING, NOT_ATTENDING, TENTATIVE, NON_RESPONSIVE};
+
+    public RSVP getUserRSVP(String userId) {
```

Comment:
a. this should appear later in the code (after the constructor, at least)
b. it's ok but i would write it like this:

```
    public RSVP getUserRSVP(String userId) {
        if (attending.containsKey(userId)) {
            return RSVP.ATTENDING;
        }
        if (tentatives.containsKey(userId)) {
            return RSVP.TENTATIVE;
        }
        if (notAttending.containsKey(userId)) {
            return RSVP.NOT_ATTENDING;
        }
        if (nonResponsive.containsKey(userId)) {
            return RSVP.NON_RESPONSIVE;
        }
        return null;
    }
```

(single-line if's should still have curly braces)

Reply:
fixed